download-basket/src/main/java/downloadbasket/actions/DownloadInfo.java

```diff
@@ -0,0 +1,45 @@
+package downloadbasket.actions;
+
+import fi.nls.oskari.annotation.OskariActionRoute;
+import fi.nls.oskari.control.ActionException;
+import fi.nls.oskari.control.ActionHandler;
+import fi.nls.oskari.control.ActionParameters;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.util.ResponseHelper;
+import downloadbasket.helpers.SendDownloadDetailsToEmailThread;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+/**
+ * Checks all download- and user details of the download basket when starting the download process.
+ * Returns boolean "success".
+ */
+
+@OskariActionRoute("DownloadInfo")
+public class DownloadInfo extends ActionHandler {
+
+	private final Logger LOGGER = LogFactory.getLogger(DownloadInfo.class);
+
+	private static final String PARAM_DOWNLOAD_DETAILS = "downloadDetails";
+	private static final String PARAM_USER_DETAILS = "userDetails";
+
+	@Override
+	public void handleAction(final ActionParameters params) throws ActionException {
+
+		JSONObject job = new JSONObject();
+		String downloadDetails = params.getHttpParam(PARAM_DOWNLOAD_DETAILS);
+		String strUserDetails = params.getHttpParam(PARAM_USER_DETAILS);
+
+		try {
+			JSONObject userDetails = new JSONObject(strUserDetails);
+			JSONArray ddArray = new JSONArray(downloadDetails);
+			new SendDownloadDetailsToEmailThread(ddArray, userDetails, params.getLocale()).start();
```

Comment:
I'm still against creating and starting a new Thread here for _every_ request. Since you do not care whether or not sending the email succeeds (you're always responding with {"success": true}) you could just send that response to the client and run the Email sending part with the current thread afterwards.

I think the only thing you'd have to change is  reorganize the lines in this function so that the success response is sent and flushed to the client after you've validated that `userDetails` and `ddArray` are valid JSONObject and JSONArray. Then change `SendDownloadDetailsToEmailThread` to extend `Runnable` instead of `Thread` and call .run() instead of .start() after creating the object.